lms/services/user.py

```diff
@@ -34,11 +34,33 @@ def store_lti_user(self, lti_user: LTIUser):
             existing_user.roles = new_user.roles
 
         else:
-            self.db_session.add(new_user)
+            self._db.add(new_user)
+
+    @lru_cache
+    def get(self, application_instance, user_id: str) -> User:
```

Comment:
`user_id` here is `User.user_id`, we pass this instead of a `lti_user` as we will use this with both a `lti_user` (where we can do `lti_user.user_id`) an also while grading and in that case the `user_id` is all we have (and we don't really want to construct a fake LTIUser for this)